src/lib/shady-render.ts

```diff
@@ -33,20 +43,106 @@ const shadyTemplateFactory = (scopeName: string) =>
       let template = templateCache.get(result.strings);
       if (template === undefined) {
         const element = result.getTemplateElement();
-
-        if (typeof window.ShadyCSS === 'object') {
-          window.ShadyCSS.prepareTemplate(element, scopeName);
-        }
-
+        window.ShadyCSS.prepareTemplateDom(element, scopeName);
         template = new Template(result, element);
         templateCache.set(result.strings, template);
       }
       return template;
     };
 
+
+const TEMPLATE_TYPES = ['html', 'svg'];
+
+/**
+ * Removes all style elements from Templates for hte given scopeName
+ */
+function removeStylesFromLitTemplates(scopeName: string) {
+  TEMPLATE_TYPES.forEach((type) => {
+    const templates = templateCaches.get(getTemplateKey(type, scopeName));
+    if (templates) {
+      templates.forEach((template) => {
+        const {element: {content}} = template;
+        const styles = content.querySelectorAll('style');
+        removeNodesFromTemplate(template, new Set(Array.from(styles)));
+      });
+    }
+  });
+}
+
+const shadyRenderSet = new Set<string>();
+
+/**
+ * For the given scope name, ensures that ShadyCSS style scoping is performed.
+ * This is done just once per scope name so the fragment and template cannot
+ * be modified.
+ * (1) extracts styles from the rendered fragment and hands them to ShadyCSS
+ * to be scoped and appended to the document
+ * (2) removes style elements from all lit-html Templates for this scope name.
+ */
+const ensureStylesScoped =
+    (fragment: DocumentFragment, template: Template, scopeName: string) => {
+      // only scope element template once per scope name
```

Comment:
style super-nit: Let's standardize on capitalizing the comments

Reply:
done